substrate/frame/multisig/src/lib.rs

```diff
@@ -168,6 +168,9 @@ pub mod pallet {
 
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+
+		/// Provider for the block number. Normally this is the `frame_system` pallet.
```

Comment:
It is quite unfortunate that we cannot add this to `parachain_system` and be done with it. The underlying issue is that all of our pallets rely on `frame_system`, not `parachain_system`. In other words, I pallets are written mostly with the intention to be used as a solochain, which in retrospect is not how it should have been.

TLDR, adding the type to all pallets is as good as we can do this. 

wdyt about marking `frame_system::block_number` as deprecated and explicitly asking everyone to decide on their block number source?

Reply:
> wdyt about marking frame_system::block_number as deprecated and explicitly asking everyone to decide on their block number source?

If this is something that you agree to, please make an issue for it.

Reply:
> > wdyt about marking frame_system::block_number as deprecated and explicitly asking everyone to decide on their block number source?
> 
> If this is something that you agree to, please make an issue for it.

Done: https://github.com/paritytech/polkadot-sdk/issues/6245